docs/proposals/karmadactl/Karmadactl Supports Configuration File Method for Production-Grade Installation and Deployment.md

```diff
@@ -0,0 +1,303 @@
+- Networking Options
+- Image Options
+
+Therefore, I think the configuration file can be designed to focus on these parts, structuring the YAML accordingly, and incorporating the existing configuration items into them. Additionally, similar to the commonly used `Extra Arg` field in Kubernetes structures, an extra field can be added to store other general fields. When designing the fields, it is advisable to refer to the field names in the kubeadm configuration file as much as possible, to facilitate user familiarity and quick adoption.
+
+After completing the design for reading config files in `karmada init`, we can extend the functionality of the `karmadactl config image list` command to read the required image configuration from the config file by reusing the config file reading method.
+
+### User Stories (Optional)
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### CLI flags changes
+
+This proposal proposes new flags `config`in `karmadactl init` flag set.
+
+| name   | shorthand | default | usage                  |
+| ------ | :-------: | ------- | ---------------------- |
+| config |     /     | ""      | Karmada init file path |
+
+With these flag, we will:
+
+- When the user has a karmadactl init file, they can use the `--config` flag to specify that `karmadactl init` should read the configuration from the specified config file. Additionally, external cmd flags can still be used. In the design, if there are overlapping parts between the two, the configuration file configuration items will take higher priority.
+
+
+### Adding New Commands
+
+#### karmadactl config image list
+
+- description:Shows information about the images required for Karmada deployment.
+- usage scope:Default image and configure the specified image
+- scope-related flag set:
+
+| name                      | shorthand | default | usage                                                        |
+| ------------------------- | --------- | ------- | ------------------------------------------------------------ |
+| private-image-registry    | /         | ""      | Private image registry where pull images from. If set, all required images will be downloaded from it, it would be useful in offline installation scenarios. |
+| kube-image-mirror-country | /         | ""      | Country code of the kube image registry to be used. For Chinese mainland users, set it to cn |
+
+- performance：
+  1. When the `private-image-registry` and `kube-image-mirror-country` flags are empty, the `list` command prints the images required for deploying Karmada under default settings, using the default image repository `registry.k8s.io`.
```

Comment:
In my design, the current default is to display the images of all components in karmada init. I noticed that the karmada-descheduler plugin and other plugins belong to the karmada addons command. I think it would be beneficial to add a new cmd flag to the config image in the future, to display the installation addresses of these plugin images.

Reply:
get it, thanks